metabase/metabase
# Add setup admin invitation

Epic: https://github.com/metabase/metabase/issues/19247

- Adds a way to invite an additional admin during the setup for database creation
- Converts setup to TypeScript

<img width="716" alt="Screenshot 2021-12-20 at 14 22 36" src="https://user-images.githubusercontent.com/8542534/146760026-0989dead-b74b-4ecd-8dc2-2fb226ba4eca.png">
<img width="716" alt="Screenshot 2021-12-20 at 14 22 51" src="https://user-images.githubusercontent.com/8542534/146760032-1b4ec133-c8f1-47ea-912a-8fd7d61637af.png">
<img width="737" alt="Screenshot 2021-12-20 at 14 22 56" src="https://user-images.githubusercontent.com/8542534/146760033-32f14f34-b1fe-4c93-a6d3-f2743b30ec1e.png">

How to test:
- `docker run -p 80:80 -p 25:25 maildev/maildev`
- `MB_EMAIL_SMTP_HOST=localhost MB_EMAIL_SMTP_PORT=25 MB_EMAIL_SMTP_USERNAME=admin MB_EMAIL_SMTP_PASSWORD=admin MB_EMAIL_SMTP_SECURITY=none MB_EMAIL_EMAIL_FROM_ADDRESS=mailer@metabase.test yarn dev` with a fresh Metabase instance
- Go to the database step. There should be a new section for inviting a user for database setup.
- Make sure that these details are sent to the backend in `POST /api/setup` call. They are currently unused so nothing more can be checked.
